New frontend with tailwind

OmTheTurtle opened the pull request:
Closes #443 #140 

It's finally here (almost). So there are a small amount of features missing (no dark theme for datetime picker, no side-calendar when creating a new group), but I feel like I already worked enough on this issue to prepare it for release. I'll deploy it to the staging env ([https://tanulo.kir-dev.sch.bme.hu](https://tanulo.kir-dev.sch.bme.hu)) so everyone can try and give feedback easily.

poiuztrewq1 reviewed (approved):
Looks really good, I like it.

OmTheTurtle commented on views/group/new.pug:
```diff
@@ -2,80 +2,68 @@ extends ../layouts/main
-    .field
-      .control
+                option(value=`${room}`)= room
+          button(type="button", class="px-2 py-1 border border-gray-600 rounded-md hover:border-gray-500", @click="open = true") Naptár
```
Right, I forgot to comment this out too.

OmTheTurtle commented on views/group/new.pug:
```diff
@@ -2,80 +2,68 @@ extends ../layouts/main
+      div
+        label(class='text-lg font-bold' for="maxAttendees") Maximális létszám
+        br
+        input(type="number" name="maxAttendees" id="maxAttendees" placeholder="Alapértelmezetten 100" value=`${maxAttendees ? maxAttendees : ''}` min=1 max=100 class="rounded form-element")
```
I'll change the placeholder text, but changing the default should be a separate PR.

OmTheTurtle commented on views/group/index.pug:
```diff
@@ -1,46 +1,54 @@
+            svg(xmlns='http://www.w3.org/2000/svg' fill='none' viewbox='0 0 24 24' stroke='currentColor' class='w-6 h-6')
+              path(stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4')
+            span(class='text-lg sm:text-xl') #{group.room}. emelet
+          div(class='ml-8 break-words') !{group.description}
```
The @tailwind base; plugin has some default styles that remove list / heading styling, but using the typography plugin it can be resolved.

OmTheTurtle commented on public/js/main.js:
```diff
@@ -4,31 +4,10 @@ const UNAUTHORIZED_MESSAGE = 'Ehhez a művelethez bejelentkezés szükséges'
+function displayMessage(message) {
   const errorEl = document.createElement('article')
-  errorEl.classList.add('message', 'alert-detail', `is-${type}`)
+  errorEl.classList.add('p-6', 'text-gray-900', 'bg-red-100', 'border-l-4', 'border-red-500')
```
I think I'll assert this is a future issue.

balintkiraly commented on public/js/themes.js:
```diff
@@ -1,47 +1,26 @@
+
+  } else {
+    document.cookie = 'theme=dark;path=/;SameSite=Lax'
   }
```
I guess you can solve this in one line instead of the if
```
 document.cookie = `theme=${getCookie('theme')};path=/;SameSite=Lax`
 ```

OmTheTurtle replied:
The purpose of this is to change from light to dark, so i don't think that approach will work. I refactored to something that uses a clearer style.

balintkiraly replied:
You're right, sorry

falusi94 reviewed (commented):
Good job! Long PR so I mainly checked the staging to gather some feedback.

**1. Spacing**
I feel the spacing on some pages weird. For example on the home page the list items have a huge top margin comparing to other elements, decreasing the margin to `calc(1.5rem * calc(1 - var(--tw-space-y-reverse)))` feels much more natural. Though I'm not so familiar with tailwind to tell how to address this issue. Similar thoughts on the groups page.
 
<img width="1155" alt="Screenshot 2021-03-10 at 20 43 37" src="https://user-images.githubusercontent.com/10588147/110695073-521a1d80-81e1-11eb-9709-7db2ea7b2703.png">

**2. Colours**
There is some selection of colours that do not have enough contrast in my opinion. On the tickets page there is this notification that has a background almost same as the background behind it.

<img width="744" alt="Screenshot 2021-03-09 at 21 57 16" src="https://user-images.githubusercontent.com/10588147/110693679-9b696d80-81df-11eb-8b7d-c76625a65f9a.png">

Also the rounded yellowish/pinkish buttons with the black text on the bluish background.

OmTheTurtle commented:
Hey @falusi94 thank you very much for the feedback.I fixed the spacing / color contrast issues, and fixed some other behavior that I didn't catch before with the datetimepicker, and also cleaned up some dependencies.
